src/meta/src/barrier/mod.rs

```diff
@@ -794,17 +920,60 @@ impl GlobalBarrierManager {
     /// Handle the new barrier from the scheduled queue and inject it.
     fn handle_new_barrier(&mut self, scheduled: Scheduled) -> MetaResult<()> {
         let Scheduled {
-            command,
+            mut command,
             mut notifiers,
             send_latency_timer,
             checkpoint,
             span,
         } = scheduled;
 
-        let (pre_applied_graph_info, pre_applied_subscription_info) =
-            self.state.apply_command(&command);
+        if let Some(table_to_cancel) = command.table_to_cancel()
+            && self
+                .checkpoint_control
+                .creating_streaming_job_controls
+                .contains_key(&table_to_cancel)
+        {
+            warn!(
+                table_id = table_to_cancel.table_id,
+                "ignore cancel command on creating streaming job"
+            );
+            for notifier in notifiers {
+                notifier
+                    .notify_start_failed(anyhow!("cannot cancel creating streaming job, the job will continue creating until created or recovery").into());
+            }
+            return Ok(());
+        }
 
         let (prev_epoch, curr_epoch) = self.state.next_epoch_pair();
+
+        // Insert newly added creating job
+        if let Command::CreateStreamingJob {
+            job_type: CreateStreamingJobType::SnapshotBackfill(snapshot_backfill_info),
+            info,
+        } = &command
+        {
+            self.checkpoint_control
+                .creating_streaming_job_controls
+                .insert(
+                    info.table_fragments.table_id(),
+                    CreatingStreamingJobControl::new(
+                        info.clone(),
+                        snapshot_backfill_info.clone(),
+                        prev_epoch.value(),
+                        &self.checkpoint_control.hummock_version_stats,
+                    ),
+                );
+        }
+
+        // may inject fake barrier
+        for creating_job in self
+            .checkpoint_control
+            .creating_streaming_job_controls
+            .values_mut()
+        {
+            creating_job.may_inject_fake_barrier(&mut self.control_stream_manager, checkpoint)?
```

Comment:
Do we have metrics for the barrier latency (including e2e latency, inflight latency and sync latency) for the fake barrier?